### PR TITLE
Budget fiddling

### PR DESCRIPTION
**Pull request by graydon**
A bunch of exploration of minor issues discovered in budget calibration, ostensibly about #1020 but also involving some tracy markup, some internal cleanup, machinery to allow excluding the overwhelming VM instantiation cost center, and some attempts at setting budget costs more from first principles.

### CODE REVIEW

File: soroban-env-host/benches/common/measure.rs
Comment (jayz22) [inline soroban-env-host/benches/common/measure.rs:47]: This is a nice check!

File: soroban-env-host/benches/common/measure.rs
Comment (jayz22) [inline soroban-env-host/benches/common/measure.rs:347]: Nice check here!

File: soroban-env-host/src/budget.rs
Comment (jayz22) [inline soroban-env-host/src/budget.rs:551]: I've been thinking about consolidating some of those cost types, and put some thoughts in https://github.com/stellar/rs-soroban-env/issues/1020#issuecomment-1749621087

Some of these models have different linear coefficients, e.g. `HostMemCpy` vs. `HostMemCmp` which can be reasoned, and `ValSer` and `ValDeser` which is harder to reason about. 
Also they have different const coefficients, which will probably need to come from measure. 
So I was thinking a combination of measurement and reasoning to determine the proper overhead and scaling factor for the analytical model maybe the right thing to do? 
It can be a followup, don't want to block this PR. 

Another way that could work is in the calibration, copy/compare a small bytes array N times, instead of on one big array. This could avoid the AVX optimization? Personally I still prefer the analytical approach.

File: soroban-env-host/src/budget.rs
Comment (graydon) [reply soroban-env-host/src/budget.rs:266]: I thought of this, but then I also thought that part of why we got rid of it is that it's a bit .. recursive? like if you were to start counting the couple-hundred instructions spent, probably a lot of them are in the steps _before_ we get to this line: updating the tracker, fetching the CPU cost model and evaluating it (and then doing that again for memory), and so if we did all that a _second time_ just to "charge the cost of mesuring costs" we'd be incurring those costs again, and would need to "charge the cost of measuring costs" again, so we'd have to actually set that cost to _twice_ the measurement cost. Since this cost is not entirely negligible, I figured it might be better to just inline it inside here, where it's going to be much smaller, probably only 2-5 instructions (a branch, a saturating add, a store) and only apply to the CPU path.

File: soroban-env-host/src/budget.rs
Comment (graydon) [reply soroban-env-host/src/budget.rs:551]: Yeah, we can continue to iterate on this. I don't think this PR is the final word on the question; just what came to mind as I was doing review. I wanted to have a look at x86 (done so!) and do some eyeball-checks with tracy (done so!) and once I did those I wanted to shake some of the most obvious quirks out of what I found (memcpy numbers that don't make sense, plus the issue of overcounting VM setup obscuring undercounting the rest).

**Comment by graydon** [comment]
@jayz22 on second thought I reverted the hard-wired constant measurement overhead; it breaks tests and does so in a way that suggests why it's wrong: it breaks the implicit contract that "costs are reflected by the cost schedule only". Maybe we can just give everything in the cost schedule slightly higher constant terms to cover the measurement overhead.

**Review by dmkozh — APPROVED** [review]

**Pull request merged**
